headers/xdp/xsk.h

```diff
@@ -203,6 +204,31 @@ struct xsk_umem_config {
 	__u32 flags;
 };
 
+/* The following fields are optional:
+ * 
+ * @fd, @size, @fill_size, @comp_size, @frame_size, @frame_headroom,
+ * @flags, @tx_metadata_len
+ *  If @fd is unset, a new sockfd will be created.
+ *  If @size is unset, @umem_area must be page-aligned.
+ *  If the remaining fields are unset, they will be set to 
+ *  default value (see `xsk_set_umem_config()`).
```

Comment:
So this comment is not actually accurate in its current form. Meaning that the code doesn't actually check for "unset" fields. Rather, if the struct is passed at all, the fields should be set to their default values (fd to -1, frame_size to XSK_UMEM__DEFAULT_FRAME_SIZE, etc) to get those.

So either we should update the documentation comment, or we should change the code to actually treat 0-valued fields as meaning "give me the default". We have some precedence for the latter (e.g., `xdp_program__create()` treats fd 0 as "unset", as do most of the libbpf APIs), but it would be inconsistent with the existing `xsk_umem__create_with_fd`. So either way we will have some inconsistency I think I'm leaning towards treating 0-values as unset, because that makes things more ergonomic for users of the opts API; WDYT, @magnus-karlsson ?

Reply:
Let us treat 0-values as unset so this new API becomes consistent with the other libxdp APIs. Thanks.

Reply:
I think this is reasonable. Changes for this issue are in a new commit `d7c82d9`.

Reply:
Got it, a new commit `[124faa9](https://github.com/xdp-project/xdp-tools/commit/124faa92223bdd8515578aa8109eccbf8595d035)` is to replace the previous, please take a look!